### DISCUSSION TITLE
Is there a way to stop Poetry trying all the different overrides?

### DISCUSSION THREAD

**Discussion by kupuguy** [python-poetry/poetry#4696] [Q&A]
I have a project where dependency resolution takes about a minute but one of the dependencies (Pandas) has a complex set of dependencies so when Poetry looks at that package it starts again from scratch and tries all the different combinations of markers so the entire process takes 5 minutes instead.

From https://github.com/pandas-dev/pandas/blob/master/setup.cfg
```
install_requires =
    numpy>=1.18.5; platform_machine!='aarch64' and platform_machine!='arm64' and python_version<'3.10'
    numpy>=1.19.2; platform_machine=='aarch64' and python_version<'3.10'
    numpy>=1.20.0; platform_machine=='arm64' and python_version<'3.10'
    numpy>=1.21.0; python_version>='3.10'
    python-dateutil>=2.8.1
    pytz>=2020.1
```

In my pyproject.toml I've specified a minimum version for numpy, and I thought that might stop it retrying everything. It seems to me that by telling it that it needs numpy >=1.21.0 for all platforms that should mean the platform specific resolutions are redundant:

```
numpy=">=1.21.0"
pandas="1.3.4"
```
Sadly it makes no difference setting a version for numpy, it still goes through the entire resolution process several times.

I can limit the combination by setting Python <3.10 so it only tries the first three combinations, or >=3.10 so it only tries the last one (but I have Black as a dependency also and Black has an override for Python >=3.10 so that more than doubles the total time again).

**Comment by jaymegordo** [comment]
+1 came here after the exact same troubleshooting steps, would love know if there's a way to prevent checking all the overrides. Even explicitely locking python/pandas/numpy versions the overrides are all still tried and resolution takes 3x longer.
